### PR TITLE
📚 Add action to check broken links

### PR DESCRIPTION
**Pull request by AlexanderDokuchaev**
## 📝 Description

- Added action https://github.com/AlexanderDokuchaev/md-dead-link-check to check broken links in repository
  - "docs/" - excluded from check because files are formatted for sphinx .
  - "https://openvinotoolkit.github.io/anomalib/*" - as i understand documentation will be published later, is it?
- Added check to nightly action to check all external links.
- Fixed broken links
- Updated links to absolute path `/docs` instead of relative `../../../../docs/`, it shortly and safety if move file.
- Fixed link [Code Review Checklist](https://github.com/openvinotoolkit/anomalib/docs/source/markdown/guides/developer/code_review_checklist.md) in PR template, currently `Not Found`.
- Removed `Benchmarking and Hyperparameter Optimization` from notebooks,  it is not exist. 
- Some links to the GitHub site replaced to local paths. This is debatable, but I believe it's more useful to be able to read the documentation within a editor and then easily switch to the corresponding file instead of having to open it in a browser.

## ✨ Changes

Select what type of change your PR is:

- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔨 Refactor (non-breaking change which refactors the code base)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] 📚 Documentation update
- [ ] 🔒 Security update

## ✅ Checklist

Before you submit your pull request, please make sure you have completed the following steps:

- [x] 📋 I have summarized my changes in the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) and followed the guidelines for my type of change (skip for minor changes, documentation updates, and test enhancements).
- [x] 📚 I have made the necessary updates to the documentation (if applicable).
- [x] 🧪 I have written tests that support my changes and prove that my fix is effective or my feature works (if applicable).

For more information about code review checklists, see the [Code Review Checklist](../docs/source/markdown/guides/developer/code_review_checklist.md).

### CODE REVIEW

**Comment by AlexanderDokuchaev** [comment]
Hi @samet-akcay 
I used this tool for NNCF repository, hope it will useful for your too :wink:  
It can be used as pre-commit hook (works less than 1 second), if need i can change it.

Example of output:
```
File: README.md:262 • Link: https://github.com/openvinotoolkit/anomalib/tree/main/src/anomalib/models/patchcore#mvtec-ad-dataset • Error: 404: Not Found
File: notebooks/100_datamodules/README.md:5 • Link: 100_datamodules/101_btech.ipynb • Error: Path does not exists in repository
File: notebooks/100_datamodules/README.md:6 • Link: 100_datamodules/102_mvtec.ipynb • Error: Path does not exists in repository
File: notebooks/100_datamodules/README.md:7 • Link: 100_datamodules/103_folder.ipynb • Error: Path does not exists in repository
File: notebooks/100_datamodules/README.md:8 • Link: 100_datamodules/104_tiling.ipynb • Error: Path does not exists in repository
File: notebooks/500_use_cases/501_dobot/README.md:57 • Link: https://github.com/openvinotoolkit/anomalib/blob/main/notebooks/500_use_cases/501_dobot/501_dobot/501b_inference_with_a_robotic_arm.ipynb • Error: 404: Not Found
File: notebooks/README.md:46 • Link: 300_benchmarking/301_benchmarking.ipynb • Error: Path does not exists in repository
File: notebooks/README.md:47 • Link: 300_benchmarking/302_hpo_wandb.ipynb • Error: Path does not exists in repository
```

**Review by samet-akcay — APPROVED** [review]
thanks for fixing these @AlexanderDokuchaev! I'm fine with the changes, but will also wait for @ashwinvaidya17's review

File: pyproject.toml [hunk 1]
```diff
@@ -194,3 +194,13 @@ source = [
...
+
+# # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # # #
+# LINK CHECKER CONFIGURATION                                                  #
+[tool.md_dead_link_check]
+exclude_links = [
+    "https://github.com/openvinotoolkit/anomalib/settings/actions/runners/new",
+    "https://openvinotoolkit.github.io/anomalib/*", # Remove it after publish documentation
+]
+exclude_files = ["docs/*"]
+force_get_requests_for_links = ["https://secure.ethicspoint.com/*"]
```
Comment (ashwinvaidya17) [inline pyproject.toml:206]: What's this part?

Comment (AlexanderDokuchaev) [reply pyproject.toml:206]: The link exists in code of conduct. https://github.com/openvinotoolkit/anomalib/blob/5babb3b39290c66cf5d8210f8cae2ab1f8825ea2/CODE_OF_CONDUCT.md?plain=1#L63
This site is not supported `head` request, always return 404, need to use get request to check link.

**Comment by AlexanderDokuchaev** [comment]
Updated PR, 
- For pull_request action will check external links only in modified files.
- All links will checked in nightly action.

**Pull request merged**